src/fetch.js

```diff
@@ -33,7 +33,7 @@ export function fetchImpl(url, { method, headers, body, onNext, onError, onCompl
                 }
             });
         } else {
-            response.json().then(json => {
+            return response.json().then(json => {
```

Comment:
need to return or else errors parsing json responses will not go to the `.catch`